src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeSemantic.java

```diff
@@ -56,15 +56,15 @@ public boolean equals(int a) // wrong arg type, ok
 
     public class EqualsVsHashCode2
     {
-        public boolean equals(String a) // ok
```

Comment:
The file lgtm....
Please go through the Issue https://github.com/checkstyle/checkstyle/issues/13213 description before sending PR.
You will have to remove the respective suppression from [checkstyle-input-suppressions.xml](https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle-input-suppressions.xml)